service.go

```diff
@@ -4,95 +4,54 @@ import (
 	"context"
```

Comment:
this file has become very thin now ... should we consider putting the builder logic here as well ?

Reply:
we could!